### PR TITLE
roctracer-dev: inherit ROCmPackage, add v5.3.0, and fix external detection

### PR DESCRIPTION
**Pull request by bvanessen**
Fixed a bug in how the external is identified for the roctracer package.  Added a new version hash.  Made the package a proper ROCm package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/roctracer-dev/package.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class RoctracerDev(CMakePackage):
     maintainers = ["srekolam", "renjithravindrankannath"]
     libraries = ["libroctracer64"]
 
+    version("5.3.0", sha256="36f1da60863a113bb9fe2957949c661f00a702e249bb0523cda1fb755c053808")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/roctracer-dev/package.py]: Confirmed the sha256.

**Comment by bvanessen** [comment]
@tldahlgren @srekolam Can we get this merged soon, I have some dependent PRs on it.  Thanks.

**Comment by alalazo** [comment]
The [failure in CI](https://gitlab.spack.io/spack/spack/-/jobs/3970262) looks legit

**Comment by bvanessen** [comment]
> The [failure in CI](https://gitlab.spack.io/spack/spack/-/jobs/3970262) looks legit

@alalazo I think I see and fixed the problem.  I just pushed the change.

**Comment by tldahlgren** [comment]
There are also required package audit checks that are failing.  

Correct me if I'm wrong, but is this PR dependent on #33320?  That PR is having E4S stack build issues for several packages.

**Comment by bvanessen** [comment]
> There are also required package audit checks that are failing.
> 
> Correct me if I'm wrong, but is this PR dependent on #33320? That PR is having E4S stack build issues for several packages.

@tldahlgren Sort of.  It has some redundancy in adding the 5.3.0 hashes, but the actual bug fix is orthogonal.  Let me back out the 5.3.0 hash values so that it can merge with #33320 when that passes.

**Comment by bvanessen** [comment]
> > There are also required package audit checks that are failing.
> > Correct me if I'm wrong, but is this PR dependent on #33320? That PR is having E4S stack build issues for several packages.
> 
> @tldahlgren Sort of. It has some redundancy in adding the 5.3.0 hashes, but the actual bug fix is orthogonal. Let me back out the 5.3.0 hash values so that it can merge with #33320 when that passes.

@tldahlgren I think that this should now be clean with respect to #33320.

**Comment by bvanessen** [comment]
@alalazo @tldahlgren @srekolam Can this be merged?

**Comment by bvanessen** [comment]
@spackbot run pipeline

**Comment by bvanessen** [comment]
@alalazo @tldahlgren @srekolam I have rebased now that 5.3.0 versions have been added.

**Comment by bvanessen** [comment]
@spackbot run pipeline

**Comment by bvanessen** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]
Trusting @bvanessen on the detection changes. Using a base class LGTM.

**Pull request merged**